src/libs/relay/conduit_relay_io_silo.cpp

```diff
@@ -573,900 +420,4343 @@ conduit_wedge_connectivity_to_silo(Node &n_mesh_conn)
+void
+read_quadmesh_domain(DBquadmesh *quadmesh_ptr,
+                     const std::string &multimesh_name,
+                     conduit::Node &mesh_domain)
 {
-    // also support interleaved:
-    // xy, xyz 
-    // convert these to separate coord arrays for silo 
-    const Node &n_coord_vals = n_coords["values"];
-
-    // check if we are 2d or 3d
-    int num_coords = 2;    
-    if(!n_coord_vals.has_path("x"))
+    int coordtype{quadmesh_ptr->coordtype};
+    int ndims{quadmesh_ptr->ndims};
+    int dims[] = {quadmesh_ptr->nnodes,
+                  quadmesh_ptr->nnodes,
+                  quadmesh_ptr->nnodes};
+    int *real_dims = dims;
+
+    if (coordtype == DB_COLLINEAR)
     {
-        CONDUIT_ERROR( "mesh coordset missing: x");
+        mesh_domain["coordsets"][multimesh_name]["type"] = "rectilinear";
+        mesh_domain["topologies"][multimesh_name]["type"] = "rectilinear";
+        real_dims = quadmesh_ptr->dims;
     }
-    if(!n_coord_vals.has_path("y"))
+    else if (coordtype == DB_NONCOLLINEAR)
     {
-        CONDUIT_ERROR( "mesh coordset missing: y");
+        mesh_domain["coordsets"][multimesh_name]["type"] = "explicit";
+        mesh_domain["topologies"][multimesh_name]["type"] = "structured";
+
+        // We subtract 1 from each of these because in silo these dims are node dims, not element dims
+        mesh_domain["topologies"][multimesh_name]["elements/dims/i"] = quadmesh_ptr->dims[0] - 1;
+        if (ndims > 1) mesh_domain["topologies"][multimesh_name]["elements/dims/j"] = quadmesh_ptr->dims[1] - 1;
+        if (ndims > 2) mesh_domain["topologies"][multimesh_name]["elements/dims/k"] = quadmesh_ptr->dims[2] - 1;
     }
-    if(n_coord_vals.has_path("z"))
+    else
     {
-        num_coords = 3;
+        CONDUIT_ERROR("Undefined coordtype in " << coordtype);
     }
 
-    const char* coordnames[3] = {"x", "y", "z"};
-
-    Node n_coords_compact;
-    // compaction is necessary to support ragged arrays
-    n_coord_vals.compact_to(n_coords_compact);
+    mesh_domain["topologies"][multimesh_name]["coordset"] = multimesh_name;
 
-    int num_pts = n_coords_compact["x"].dtype().number_of_elements();
-    // TODO: check that y & z have the same number of points
+    // If the origin is not the default value, then we need to specify it
+    if (quadmesh_ptr->base_index[0] != 0 && 
+        quadmesh_ptr->base_index[1] != 0 && 
+        quadmesh_ptr->base_index[2] != 0)
+    {
+        Node &origin = mesh_domain["topologies"][multimesh_name]["elements"]["origin"];
+        origin["i"] = quadmesh_ptr->base_index[0];
+        if (ndims > 1) origin["i"] = quadmesh_ptr->base_index[1];
+        if (ndims > 2) origin["i"] = quadmesh_ptr->base_index[2];
+    }
 
-    n_mesh_info[topo_name]["num_pts"].set(num_pts);
+    copy_point_coords(quadmesh_ptr->datatype,
+                      quadmesh_ptr->coords,
+                      ndims,
+                      real_dims,
```

Comment:
In the collinear case (rectilinear grid), we want to use the dimensions the quadmesh provides directly. In the noncollinear case (structured grid), we are setting up an explicit coordset so the dimensions for x,y,z are the same, equal to the total number of nodes.

I create the dimensions for the structured case and then create a pointer (`real_dims`) to them so that if we hit the rectilinear case, I can make it point to the quadmesh dimensions directly.

See line 673.

There maybe is a better way to do this...

Reply:
that makes sense, I think I could not see b/c of how the diffs are presented.